### PR TITLE
[SYCL] Add mode where only last command in each batch yields a host-visible event

### PR DESCRIPTION
**Pull request by smaslov-intel**
E2E test is in https://github.com/intel/llvm-test-suite/pull/765
Signed-off-by: Sergey V Maslov <sergey.v.maslov@intel.com>

### CODE REVIEW

**Review by againull — COMMENTED** [review]
Could you please update SYCL_PI_LEVEL_ZERO_DEVICE_SCOPE_EVENTS description in the documentation https://github.com/intel/llvm/blob/sycl/sycl/doc/EnvironmentVariables.md

Could you please add a feature test to llvm-test-suite and verify it with this PR.

File: sycl/doc/EnvironmentVariables.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Environment Variables
+#Environment Variables
```
Comment (pvchupin) [inline sycl/doc/EnvironmentVariables.md:1]: It breaks formatting.

File: sycl/doc/EnvironmentVariables.md [hunk 2]
```diff
@@ -27,7 +27,9 @@ compiler and runtime.
 ### `SYCL_DEVICE_ALLOWLIST`
 
 A list of devices and their driver version following the pattern:
-`BackendName:XXX,DeviceType:YYY,DeviceVendorId:0xXYZW,DriverVersion:{{X.Y.Z.W}}`.
+`BackendName:XXX,DeviceType:YYY,DeviceVendorId:0xXYZW,DriverVersion:{
```
Comment (pvchupin) [inline sycl/doc/EnvironmentVariables.md:30]: What's the purpose of this change?

**Review by pvchupin — CHANGES_REQUESTED** [review]

File: sycl/doc/EnvironmentVariables.md [hunk 3]
```diff
@@ -27,7 +27,9 @@ compiler and runtime.
...
+  { X.Y.Z.W }
+}`.
```
Comment (smaslov-intel) [inline sycl/doc/EnvironmentVariables.md:32]:
```suggestion
`BackendName:XXX,DeviceType:YYY,DeviceVendorId:0xXYZW,DriverVersion:{{X.Y.Z.W}}`.
}`.
```

File: sycl/doc/EnvironmentVariables.md (hunk 2 above)
Comment (smaslov-intel) [reply sycl/doc/EnvironmentVariables.md]: ask clang-format )

File: sycl/doc/EnvironmentVariables.md (hunk 3 above)
Comment (smaslov-intel) [inline sycl/doc/EnvironmentVariables.md]:
```suggestion
`BackendName:XXX,DeviceType:YYY,DeviceVendorId:0xXYZW,DriverVersion:{{X.Y.Z.W}}`.
```

File: sycl/doc/EnvironmentVariables.md (hunk 1 above)
Comment (smaslov-intel) [inline sycl/doc/EnvironmentVariables.md:1]:
```suggestion
# Environment Variables
```

**Review by pvchupin — APPROVED** [review]
doc ok

**Review by againull — APPROVED** [review]

**Pull request merged**